lispy.c

```diff
@@ -1439,13 +1446,17 @@ int main(int argc, char** argv){
     lenv_add_builtins(e);
     lval* standard = lval_add(lval_sexpr(), lval_str("std-library/std_library.lspy"));
     lval* std = builtin_load(e, standard);
+    if (atexit(cleanup)) {
```

Comment:
doesn't this trigger the conditional when atexit is called correctly? It should be called in case of error

Reply:
Actually tried the change and works correctly, could you still explain to me how it does exactly work and how to trigger the error message?
Really grateful for your help, you are really knowledgeable

Reply:
According to the atexit(3) [1] the atexit function would return zero in the case of success and non zero in the case of error.

[1] http://man7.org/linux/man-pages/man3/atexit.3.html

Reply:
Ok I read the opposite thing in a forum and clearly was wrong, thanks for the resources

Reply:
Your welcome :)